File: code/1041/1041.js
const input = require("fs").readFileSync("/dev/stdin", "utf8");
const lines = input.split(/\n|\s/);

const x = parseFloat(lines[0]);
const y = parseFloat(lines[1]);

if (x === 0 && y === 0) {
  console.log("Origem");
} else if (x === 0) {
  console.log("Eixo Y");
} else if (y === 0) {
  console.log("Eixo X");
} else if (x > 0 && y > 0) {
  console.log("Q1");
} else if (x > 0 && y < 0) {
  console.log("Q4");
} else if (x < 0 && y > 0) {
  console.log("Q2");
} else {
  console.log("Q3");
}

File: code/1041/1041.rb
# frozen_string_literal: true

inputs = gets.split(' ')

x = inputs[0].to_f
y = inputs[1].to_f

if x.zero? && y.zero?
  puts 'Origem'
elsif x.zero?
  puts 'Eixo Y'
elsif y.zero?
  puts 'Eixo X'
else
  puts 'Q1' if x.positive? && y.positive?
  puts 'Q2' if x.negative? && y.positive?
  puts 'Q3' if x.negative? && y.negative?
  puts 'Q4' if x.positive? && y.negative?
end
